File: Java/Hussain Sir Java/Fundamentals/Language/Basics/Interval.java
class Interval{

	private int min;
	private int sec;

	public Interval(int m, int s){
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes(){
		return min;
	}

	public int seconds(){
		return sec;
	}

	public int time(){
		return 60 * min + sec;
	}

	public void print(){
		System.out.printf("%d:%02d%n", min, sec);
	}
}

File: Fundamentals/Language/Inheritance/interval.cs
using System;

//a part of the definition of this class can appear in other source file
partial class Interval
{
	public int Minutes {get;}

	public int Seconds {get;}

	partial void OnCreate();

	public Interval(int minutes, int seconds)
	{
		Minutes = minutes + seconds / 60;
		Seconds = seconds % 60;
		OnCreate();
	}

	public int GetTime()
	{
		return 60 * Minutes + Seconds;
	}

	public override string ToString()
	{
		return $"{Minutes}:{Seconds:00}";
	}

	public override int GetHashCode() => Minutes + Seconds;
	
	public override bool Equals(object other)
	{
		if(other is Interval that)
		{
			return this.Minutes == that.Minutes && this.Seconds == that.Seconds;
		}

		return false;
	}
}

